### PR TITLE
member selector support complex member path

### PR DESCRIPTION
**Pull request by donandren**
member selector support complex member path like ChildProperty.Value

### CODE REVIEW

**Comment by grokys** [comment]
Looks good! Would you have time to write a few tests for this?

**Comment by donandren** [comment]
sure :)

**Comment by grokys** [comment]
Excellent, thanks! Looks like you found some problems with your patch when writing the tests, too!

I will review and merge shortly!

File: src/Markup/Avalonia.Markup.Xaml/Templates/MemberSelector.cs [hunk 1]
```diff
@@ -1,20 +1,64 @@
...
+            {
+                return null;
+            }
+
+            return result;
         }
+
+        private ExpressionNode _expressionNode;
+        private string _memberName;
+        private ExpressionNode _memberValueNode;
```
Comment (grokys) [inline src/Markup/Avalonia.Markup.Xaml/Templates/MemberSelector.cs]: We usually declare fields at the top of the class.

File: src/Markup/Avalonia.Markup.Xaml/Templates/MemberSelector.cs [hunk 2]
```diff
@@ -1,20 +1,64 @@
...
+            }
+
+            if (_expressionNode == null)
+            {
+                _expressionNode = ExpressionNodeBuilder.Build(MemberName);
+
+                _memberValueNode = _expressionNode;
+
+                while (_memberValueNode.Next != null)
+                    _memberValueNode = _memberValueNode.Next;
```
Comment (grokys) [inline src/Markup/Avalonia.Markup.Xaml/Templates/MemberSelector.cs]: Put braces around loop bodies.

**Comment by grokys** [comment]
Merged! I added a couple of small style nits but don't worry about them - I'll fix them - it's just for future reference.

**Pull request merged**